src/middleware.ts

```diff
@@ -10,6 +10,10 @@ acceptLanguage.languages(locales);
 export function middleware(request: NextRequest) {
   const response = i18nRouter(request, i18nConfig);
 
+  // Set a cookie with the pathname that was used on the first page load
```

Comment:
why this @chybisov ?

Reply:
@tcheee we can't access pathname inside the server components, so it's a workaround :)